mingw-w64-rust/config.toml

```
@@ -115,3 +115,10 @@ default-linker = "wasm-ld.exe"
 sanitizers = false
 profiler = false
 wasi-root = "$MINGW_PREFIX/share/wasi-sysroot"
+
+[target.x86_64-pc-cygwin]
+cc = "$MSYS2_PREFIX/bin/gcc.exe"
+cxx = "$MSYS2_PREFIX/bin/g++.exe"
+ar = "$MSYS2_PREFIX/bin/ar.exe"
+ranlib = "$MSYS2_PREFIX/bin/ranlib.exe"
+linker = "x86_64-pc-msys-gcc.exe"
```

```suggestion
linker = "x86_64-pc-msys-gcc.exe"
default-linker = "x86_64-pc-msys-gcc.exe"
```
default-linker is for rustc runtime, linker is for rustc build time

---

Fixed.